{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Análise de sequências e features no NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import de módulos e bibliotecas\n",
    "\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Funções para aceder, extraír e guardar informação de sequências do NCBI nomeadamente as suas annotations, features e qualifiers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "Entrez.email = input(\"Insere o teu e-mail\") \n",
    "def access_ncbi_seq(accession : str, database : str, format : str, filename : str):\n",
    "    \"\"\"\n",
    "    função para aceder, extraír e guardar a informação de sequências do NCBI\n",
    "    recebe o accession id da sequência, a respetiva base de dados ('nucleotide' ou 'protein'),\n",
    "    o formato em que queremos trabalhar a informação ('gb' ou 'faa') e o nome que queremos dar ao ficheiro\n",
    "    \"\"\"    \n",
    "    handle = Entrez.efetch(db=database, id=accession, rettype=format, retmode=\"text\")\n",
    "    record = SeqIO.read(handle, format)\n",
    "    handle.close()\n",
    "\n",
    "    SeqIO.write(record, filename, format)\n",
    "\n",
    "    record = SeqIO.read(filename, format)\n",
    "\n",
    "    return record\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "def seq_annotations(record):\n",
    "    \"\"\"\n",
    "    função que mostra as anotações do ficheiro genbank criado\n",
    "    recebe o record obtido ao ler o ficheiro através da função read_genbank_file\n",
    "    \"\"\"\n",
    "    print(\"ID:\", record.id)\n",
    "    print(\"\\nName:\", record.name)\n",
    "    print(\"\\nDescription:\", record.description)\n",
    "    print(\"\\nSequence length:\", len(record))\n",
    "    print(\"\\nGeneral annotations:\")\n",
    "    for annotation, content in record.annotations.items():\n",
    "          print(f\">{annotation}: {content}\")\n",
    "    if len(record.dbxrefs) != 0:\n",
    "        print(\"External Databases References:\", record.dbxrefs) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "def seq_features_qualifiers(record):\n",
    "    \"\"\"\n",
    "    função que mostra as features e os seus qualifiers\n",
    "    recebe o record obtido ao ler o ficheiro através da função read_genbank_file\n",
    "    \"\"\"\n",
    "    print(len(record.features), \"features\\n\")\n",
    "    print(\"Type and Location:\")\n",
    "    for feature in record.features:\n",
    "        print(f\">{feature.type}, {feature.location}\")\n",
    "    print(\"\\nQualifiers:\")\n",
    "    for k in range(len(record.features)):\n",
    "        print(f\">{record.features[k].qualifiers}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene FLG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Análise da sequência de mRNA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Anotações da sequência de mRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: NM_002016.2\n",
      "\n",
      "Name: NM_002016\n",
      "\n",
      "Description: Homo sapiens filaggrin (FLG), mRNA\n",
      "\n",
      "Sequence length: 12793\n",
      "\n",
      "General annotations:\n",
      ">molecule_type: mRNA\n",
      ">topology: linear\n",
      ">data_file_division: PRI\n",
      ">date: 22-JAN-2024\n",
      ">accessions: ['NM_002016']\n",
      ">sequence_version: 2\n",
      ">keywords: ['RefSeq', 'MANE Select']\n",
      ">source: Homo sapiens (human)\n",
      ">organism: Homo sapiens\n",
      ">taxonomy: ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      ">references: [Reference(title='Temporal relationships between Staphylococcus aureus colonization, filaggrin expression, and pediatric atopic dermatitis', ...), Reference(title='Filaggrin gene polymorphisms in Indian children with atopic dermatitis: A cross-sectional multicentre study', ...), Reference(title='Filaggrin gene variants among Saudi patients with ichthyosis vulgaris', ...), Reference(title='Deep palmar phenotyping in atopic eczema: patterns associated with filaggrin variants, disease severity and barrier function in a South Asian population', ...), Reference(title='Skin Barrier Function and Its Relationship With IL-17, IL-33, and Filaggrin in Malar Melasma', ...), Reference(title='Characterization of the human epidermal profilaggrin gene. Genomic organization and identification of an S-100-like calcium binding domain at the amino terminus', ...), Reference(title='Organization, structure, and polymorphisms of the human profilaggrin gene', ...), Reference(title='Organization, structure, and polymorphisms of the human profilaggrin gene', ...), Reference(title='Characterization of a cDNA clone encoding human filaggrin and localization of the gene to chromosome region 1q21', ...), Reference(title='Ichthyosis vulgaris: identification of a defect in synthesis of filaggrin correlated with an absence of keratohyaline granules', ...)]\n",
      ">comment: REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from AL356504.5.\n",
      "On Aug 31, 2019 this sequence version replaced NM_002016.1.\n",
      "Summary: The protein encoded by this gene is an intermediate\n",
      "filament-associated protein that aggregates keratin intermediate\n",
      "filaments in mammalian epidermis. It is initially synthesized as a\n",
      "polyprotein precursor, profilaggrin (consisting of multiple\n",
      "filaggrin units of 324 aa each), which is localized in keratohyalin\n",
      "granules, and is subsequently proteolytically processed into\n",
      "individual functional filaggrin molecules. Mutations in this gene\n",
      "are associated with ichthyosis vulgaris.[provided by RefSeq, Dec\n",
      "2009].\n",
      "Sequence Note: The RefSeq transcript and protein were derived from\n",
      "genomic sequence to make the sequence consistent with the reference\n",
      "genome assembly. The genomic coordinates used for the transcript\n",
      "record were based on alignments.\n",
      "Publication Note:  This RefSeq record includes a subset of the\n",
      "publications that are available for this gene. Please see the Gene\n",
      "record to access additional publications.\n",
      "COMPLETENESS: complete on the 3' end.\n",
      ">structured_comment: OrderedDict([('Evidence-Data', OrderedDict([('Transcript exon combination', 'L01090.1 [ECO:0000332]')])), ('RefSeq-Attributes', OrderedDict([('MANE Ensembl match', 'ENST00000368799.2/ ENSP00000357789.1'), ('RefSeq Select criteria', 'based on single protein-coding transcript')]))])\n"
     ]
    }
   ],
   "source": [
    "seq_annotations(access_ncbi_seq(\"1732746205\", \"nucleotide\", \"gb\", \"flg_mrna\")) # annotations flg_mrna"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Features e qualifiers da sequência de mRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10 features\n",
      "\n",
      "Type and Location:\n",
      ">source, [0:12793](+)\n",
      ">gene, [0:12793](+)\n",
      ">exon, [0:51](+)\n",
      ">exon, [51:210](+)\n",
      ">CDS, [72:12258](+)\n",
      ">misc_feature, [345:708](+)\n",
      ">misc_feature, [834:11985](+)\n",
      ">exon, [210:12793](+)\n",
      ">regulatory, [12772:12778](+)\n",
      ">polyA_site, [12792:12793](+)\n",
      "\n",
      "Qualifiers:\n",
      ">OrderedDict([('organism', ['Homo sapiens']), ('mol_type', ['mRNA']), ('db_xref', ['taxon:9606']), ('chromosome', ['1']), ('map', ['1q21.3'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['filaggrin']), ('db_xref', ['GeneID:2312', 'HGNC:HGNC:3748', 'MIM:135940'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('inference', ['alignment:Splign:2.1.0'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('inference', ['alignment:Splign:2.1.0'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['epidermal filaggrin']), ('codon_start', ['1']), ('product', ['filaggrin']), ('protein_id', ['NP_002007.1']), ('db_xref', ['CCDS:CCDS30860.1', 'GeneID:2312', 'HGNC:HGNC:3748', 'MIM:135940']), ('translation', ['MSTLLENIFAIINLFKQYSKKDKNTDTLSKKELKELLEKEFRQILKNPDDPDMVDVFMDHLDIDHNKKIDFTEFLLMVFKLAQAYYESTRKENLPISGHKHRKHSHHDKHEDNKQEENKENRKRPSSLERRNNRKGNKGRSKSPRETGGKRHESSSEKKERKGYSPTHREEEYGKNHHNSSKKEKNKTENTRLGDNRKRLSERLEEKEDNEEGVYDYENTGRMTQKWIQSGHIATYYTIQDEAYDTTDSLLEENKIYERSRSSDGKSSSQVNRSRHENTSQVPLQESRTRKRRGSRVSQDRDSEGHSEDSERHSGSASRNHHGSAWEQSRDGSRHPRSHDEDRASHGHSADSSRQSGTRHAETSSRGQTASSHEQARSSPGERHGSGHQQSADSSRHSATGRGQASSAVSDRGHRGSSGSQASDSEGHSENSDTQSVSGHGKAGLRQQSHQESTRGRSGERSGRSGSSLYQVSTHEQPDSAHGRTGTSTGGRQGSHHEQARDSSRHSASQEGQDTIRGHPGSSRGGRQGSHHEQSVNRSGHSGSHHSHTTSQGRSDASHGQSGSRSASRQTRNEEQSGDGTRHSGSRHHEASSQADSSRHSQVGQGQSSGPRTSRNQGSSVSQDSDSQGHSEDSERWSGSASRNHHGSAQEQSRDGSRHPRSHHEDRAGHGHSADSSRKSGTRHTQNSSSGQAASSHEQARSSAGERHGSRHQLQSADSSRHSGTGHGQASSAVRDSGHRGSSGSQATDSEGHSEDSDTQSVSGHGQAGHHQQSHQESARDRSGERSRRSGSFLYQVSTHKQSESSHGWTGPSTGVRQGSHHEQARDNSRHSASQDGQDTIRGHPGSSRRGRQGSHHEQSVDRSGHSGSHHSHTTSQGRSDASRGQSGSRSASRTTRNEEQSRDGSRHSGSRHHEASSHADISRHSQAGQGQSEGSRTSRRQGSSVSQDSDSEGHSEDSERWSGSASRNHRGSAQEQSRHGSRHPRSHHEDRAGHGHSADSSRQSGTPHAETSSGGQAASSHEQARSSPGERHGSRHQQSADSSRHSGIPRRQASSAVRDSGHWGSSGSQASDSEGHSEESDTQSVSGHGQDGPHQQSHQESARDWSGGRSGRSGSFIYQVSTHEQSESAHGRTRTSTGRRQGSHHEQARDSSRHSASQEGQDTIRAHPGSRRGGRQGSHHEQSVDRSGHSGSHHSHTTSQGRSDASHGQSGSRSASRQTRKDKQSGDGSRHSGSRHHEAASWADSSRHSQVGQEQSSGSRTSRHQGSSVSQDSDSERHSDDSERLSGSASRNHHGSSREQSRDGSRHPGFHQEDRASHGHSADSSRQSGTHHTESSSHGQAVSSHEQARSSPGERHGSRHQQSADSSRHSGIGHRQASSAVRDSGHRGSSGSQVTNSEGHSEDSDTQSVSAHGQAGPHQQSHKESARGQSGESSGRSRSFLYQVSSHEQSESTHGQTAPSTGGRQGSRHEQARNSSRHSASQDGQDTIRGHPGSSRGGRQGSYHEQSVDRSGHSGYHHSHTTPQGRSDASHGQSGPRSASRQTRNEEQSGDGSRHSGSRHHEPSTRAGSSRHSQVGQGESAGSKTSRRQGSSVSQDRDSEGHSEDSERRSESASRNHYGSAREQSRHGSRNPRSHQEDRASHGHSAESSRQSGTRHAETSSGGQAASSQEQARSSPGERHGSRHQQSADSSTDSGTGRRQDSSVVGDSGNRGSSGSQASDSEGHSEESDTQSVSAHGQAGPHQQSHQESTRGQSGERSGRSGSFLYQVSTHEQSESAHGRTGPSTGGRQRSRHEQARDSSRHSASQEGQDTIRGHPGSSRGGRQGSHYEQSVDSSGHSGSHHSHTTSQERSDVSRGQSGSRSVSRQTRNEKQSGDGSRHSGSRHHEASSRADSSRHSQVGQGQSSGPRTSRNQGSSVSQDSDSQGHSEDSERWSGSASRNHLGSAWEQSRDGSRHPGSHHEDRAGHGHSADSSRQSGTRHTESSSRGQAASSHEQARSSAGERHGSHHQLQSADSSRHSGIGHGQASSAVRDSGHRGYSGSQASDSEGHSEDSDTQSVSAQGKAGPHQQSHKESARGQSGESSGRSGSFLYQVSTHEQSESTHGQSAPSTGGRQGSHYDQAQDSSRHSASQEGQDTIRGHPGPSRGGRQGSHQEQSVDRSGHSGSHHSHTTSQGRSDASRGQSGSRSASRKTYDKEQSGDGSRHSGSHHHEASSWADSSRHSLVGQGQSSGPRTSRPRGSSVSQDSDSEGHSEDSERRSGSASRNHHGSAQEQSRDGSRHPRSHHEDRAGHGHSAESSRQSGTHHAENSSGGQAASSHEQARSSAGERHGSHHQQSADSSRHSGIGHGQASSAVRDSGHRGSSGSQASDSEGHSEDSDTQSVSAHGQAGPHQQSHQESTRGRSAGRSGRSGSFLYQVSTHEQSESAHGRTGTSTGGRQGSHHKQARDSSRHSTSQEGQDTIHGHPGSSSGGRQGSHYEQLVDRSGHSGSHHSHTTSQGRSDASHGHSGSRSASRQTRNDEQSGDGSRHSGSRHHEASSRADSSGHSQVGQGQSEGPRTSRNWGSSFSQDSDSQGHSEDSERWSGSASRNHHGSAQEQLRDGSRHPRSHQEDRAGHGHSADSSRQSGTRHTQTSSGGQAASSHEQARSSAGERHGSHHQQSADSSRHSGIGHGQASSAVRDSGHRGYSGSQASDNEGHSEDSDTQSVSAHGQAGSHQQSHQESARGRSGETSGHSGSFLYQVSTHEQSESSHGWTGPSTRGRQGSRHEQAQDSSRHSASQDGQDTIRGHPGSSRGGRQGYHHEHSVDSSGHSGSHHSHTTSQGRSDASRGQSGSRSASRTTRNEEQSGDGSRHSGSRHHEASTHADISRHSQAVQGQSEGSRRSRRQGSSVSQDSDSEGHSEDSERWSGSASRNHHGSAQEQLRDGSRHPRSHQEDRAGHGHSADSSRQSGTRHTQTSSGGQAASSHEQARSSAGERHGSHHQQSADSSRHSGIGHGQASSAVRDSGHRGYSGSQASDNEGHSEDSDTQSVSAHGQAGSHQQSHQESARGRSGETSGHSGSFLYQVSTHEQSESSHGWTGPSTRGRQGSRHEQAQDSSRHSASQYGQDTIRGHPGSSRGGRQGYHHEHSVDSSGHSGSHHSHTTSQGRSDASRGQSGSRSASRTTRNEEQSGDSSRHSVSRHHEASTHADISRHSQAVQGQSEGSRRSRRQGSSVSQDSDSEGHSEDSERWSGSASRNHRGSVQEQSRHGSRHPRSHHEDRAGHGHSADRSRQSGTRHAETSSGGQAASSHEQARSSPGERHGSRHQQSADSSRHSGIPRGQASSAVRDSRHWGSSGSQASDSEGHSEESDTQSVSGHGQAGPHQQSHQESARDRSGGRSGRSGSFLYQVSTHEQSESAHGRTRTSTGRRQGSHHEQARDSSRHSASQEGQDTIRGHPGSSRRGRQGSHYEQSVDRSGHSGSHHSHTTSQGRSDASRGQSGSRSASRQTRNDEQSGDGSRHSWSHHHEASTQADSSRHSQSGQGQSAGPRTSRNQGSSVSQDSDSQGHSEDSERWSGSASRNHRGSAQEQSRDGSRHPTSHHEDRAGHGHSAESSRQSGTHHAENSSGGQAASSHEQARSSAGERHGSHHQQSADSSRHSGIGHGQASSAVRDSGHRGSSGSQASDSEGHSEDSDTQSVSAHGQAGPHQQSHQESTRGRSAGRSGRSGSFLYQVSTHEQSESAHGRAGPSTGGRQGSRHEQARDSSRHSASQEGQDTIRGHPGSRRGGRQGSYHEQSVDRSGHSGSHHSHTTSQGRSDASHGQSGSRSASRETRNEEQSGDGSRHSGSRHHEASTQADSSRHSQSGQGESAGSRRSRRQGSSVSQDSDSEAYPEDSERRSESASRNHHGSSREQSRDGSRHPGSSHRDTASHVQSSPVQSDSSTAKEHGHFSSLSQDSAYHSGIQSRGSPHSSSSYHYQSEGTERQKGQSGLVWRHGSYGSADYDYGESGFRHSQHGSVSYNSNPVVFKERSDICKASAFGKDHPRYYATYINKDPGLCGHSSDISKQLGFSQSQRYYYYE'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['propagated from UniProtKB/Swiss-Prot (P20930.3); Region: Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['propagated from UniProtKB/Swiss-Prot (P20930.3); Region: Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('inference', ['alignment:Splign:2.1.0'])])\n",
      ">OrderedDict([('regulatory_class', ['polyA_signal_sequence']), ('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['hexamer: AATAAA'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('note', ['major polyA site'])])\n"
     ]
    }
   ],
   "source": [
    "seq_features_qualifiers(access_ncbi_seq(\"1732746205\", \"nucleotide\", \"gb\", \"flg_mrna\")) # features, qualifiers flg_mrna"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Análise da sequência da proteína"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Anotações da sequência da proteína do registo NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: NP_002007.1\n",
      "\n",
      "Name: NP_002007\n",
      "\n",
      "Description: filaggrin [Homo sapiens]\n",
      "\n",
      "Sequence length: 4061\n",
      "\n",
      "General annotations:\n",
      ">topology: linear\n",
      ">data_file_division: PRI\n",
      ">date: 22-JAN-2024\n",
      ">accessions: ['NP_002007']\n",
      ">sequence_version: 1\n",
      ">keywords: ['RefSeq', 'MANE Select']\n",
      ">source: Homo sapiens (human)\n",
      ">organism: Homo sapiens\n",
      ">taxonomy: ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      ">references: [Reference(title='Temporal relationships between Staphylococcus aureus colonization, filaggrin expression, and pediatric atopic dermatitis', ...), Reference(title='Filaggrin gene polymorphisms in Indian children with atopic dermatitis: A cross-sectional multicentre study', ...), Reference(title='Filaggrin gene variants among Saudi patients with ichthyosis vulgaris', ...), Reference(title='Deep palmar phenotyping in atopic eczema: patterns associated with filaggrin variants, disease severity and barrier function in a South Asian population', ...), Reference(title='Skin Barrier Function and Its Relationship With IL-17, IL-33, and Filaggrin in Malar Melasma', ...), Reference(title='Characterization of the human epidermal profilaggrin gene. Genomic organization and identification of an S-100-like calcium binding domain at the amino terminus', ...), Reference(title='Organization, structure, and polymorphisms of the human profilaggrin gene', ...), Reference(title='Organization, structure, and polymorphisms of the human profilaggrin gene', ...), Reference(title='Characterization of a cDNA clone encoding human filaggrin and localization of the gene to chromosome region 1q21', ...), Reference(title='Ichthyosis vulgaris: identification of a defect in synthesis of filaggrin correlated with an absence of keratohyaline granules', ...)]\n",
      ">comment: REVIEWED REFSEQ: This record has been curated by NCBI staff. The\n",
      "reference sequence was derived from AL356504.5.\n",
      "On or before Feb 21, 2005 this sequence version replaced\n",
      "XP_378901.2, XP_048104.5.\n",
      "Summary: The protein encoded by this gene is an intermediate\n",
      "filament-associated protein that aggregates keratin intermediate\n",
      "filaments in mammalian epidermis. It is initially synthesized as a\n",
      "polyprotein precursor, profilaggrin (consisting of multiple\n",
      "filaggrin units of 324 aa each), which is localized in keratohyalin\n",
      "granules, and is subsequently proteolytically processed into\n",
      "individual functional filaggrin molecules. Mutations in this gene\n",
      "are associated with ichthyosis vulgaris.[provided by RefSeq, Dec\n",
      "2009].\n",
      "Sequence Note: The RefSeq transcript and protein were derived from\n",
      "genomic sequence to make the sequence consistent with the reference\n",
      "genome assembly. The genomic coordinates used for the transcript\n",
      "record were based on alignments.\n",
      "Publication Note:  This RefSeq record includes a subset of the\n",
      "publications that are available for this gene. Please see the Gene\n",
      "record to access additional publications.\n",
      ">structured_comment: OrderedDict([('Evidence-Data', OrderedDict([('Transcript exon combination', 'L01090.1 [ECO:0000332]')])), ('RefSeq-Attributes', OrderedDict([('MANE Ensembl match', 'ENST00000368799.2/ ENSP00000357789.1'), ('RefSeq Select criteria', 'based on single protein-coding transcript')]))])\n",
      ">molecule_type: protein\n"
     ]
    }
   ],
   "source": [
    "seq_annotations(access_ncbi_seq(\"60097902\", \"protein\", \"gb\", \"flg_protein\")) # annotations flg_protein"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Features e qualifiers da sequência da proteína do registo NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31 features\n",
      "\n",
      "Type and Location:\n",
      ">source, [0:4061]\n",
      ">Protein, [0:4061]\n",
      ">Region, [1:88]\n",
      ">Site, order{[1:19], [24:27], [35:37], [38:41], [68:72], [73:75], [76:80], [81:88]}\n",
      ">Site, order{[18:19], [23:24], [26:27], [31:33], [61:62], [63:64], [65:66], [67:68], [69:70], [72:73]}\n",
      ">Region, [91:212]\n",
      ">Region, [254:3971]\n",
      ">Region, [353:>590]\n",
      ">Region, [372:428]\n",
      ">Region, [564:>950]\n",
      ">Region, [696:753]\n",
      ">Region, [898:955]\n",
      ">Region, [984:>1368]\n",
      ">Region, [1021:1077]\n",
      ">Region, [1224:1279]\n",
      ">Region, [1345:1401]\n",
      ">Region, [1546:1603]\n",
      ">Region, [1606:>1826]\n",
      ">Region, [1670:1725]\n",
      ">Region, [1993:2050]\n",
      ">Region, [2196:2252]\n",
      ">Region, [2232:>2533]\n",
      ">Region, [2318:2374]\n",
      ">Region, [2642:2698]\n",
      ">Region, [<2648:>2961]\n",
      ">Region, [2966:3022]\n",
      ">Region, [<2972:>3252]\n",
      ">Region, [3290:3346]\n",
      ">Region, [3605:>3832]\n",
      ">Region, [3614:3670]\n",
      ">CDS, [0:4061]\n",
      "\n",
      "Qualifiers:\n",
      ">OrderedDict([('organism', ['Homo sapiens']), ('db_xref', ['taxon:9606']), ('chromosome', ['1']), ('map', ['1q21.3'])])\n",
      ">OrderedDict([('product', ['filaggrin']), ('note', ['epidermal filaggrin']), ('calculated_mol_wt', ['435040'])])\n",
      ">OrderedDict([('region_name', ['S-100']), ('note', ['S-100 domain, which represents the largest family within the superfamily of proteins carrying the Ca-binding EF-hand motif. Note that this S-100 hierarchy contains only S-100 EF-hand domains, other EF-hands have been modeled separately. S100 proteins are...; cd00213']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('site_type', ['other']), ('note', ['dimerization interface [polypeptide binding]']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('site_type', ['other']), ('note', ['Ca2+ binding site [ion binding]']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('region_name', ['Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite']), ('note', ['propagated from UniProtKB/Swiss-Prot (P20930.3)'])])\n",
      ">OrderedDict([('region_name', ['Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite']), ('note', ['propagated from UniProtKB/Swiss-Prot (P20930.3)'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['ser_rich_anae_1']), ('note', ['serine-rich protein; NF033849']), ('db_xref', ['CDD:411418'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['ser_rich_anae_1']), ('note', ['serine-rich protein; NF033849']), ('db_xref', ['CDD:411418'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('gene', ['FLG']), ('gene_synonym', ['ATOD2; FLG-1; FLG1']), ('coded_by', ['NM_002016.2:73..12258']), ('db_xref', ['CCDS:CCDS30860.1', 'GeneID:2312', 'HGNC:HGNC:3748', 'MIM:135940'])])\n"
     ]
    }
   ],
   "source": [
    "seq_features_qualifiers(access_ncbi_seq(\"60097902\", \"protein\", \"gb\", \"flg_protein\")) # features, qualifiers flg_protein"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Anotações da sequência da proteína do registo UniProt no NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: P20930.3\n",
      "\n",
      "Name: FILA_HUMAN\n",
      "\n",
      "Description: RecName: Full=Filaggrin\n",
      "\n",
      "Sequence length: 4061\n",
      "\n",
      "General annotations:\n",
      ">topology: linear\n",
      ">data_file_division: PRI\n",
      ">date: 08-NOV-2023\n",
      ">accessions: ['P20930']\n",
      ">sequence_version: 3\n",
      ">keywords: ['3D-structure', 'Calcium', 'Citrullination', 'Coiled coil', 'Developmental protein', 'Direct protein sequencing', 'Ichthyosis', 'Metal-binding', 'Phosphoprotein', 'Reference proteome', 'Repeat']\n",
      ">source: Homo sapiens (human)\n",
      ">organism: Homo sapiens\n",
      ">taxonomy: ['Eukaryota', 'Metazoa', 'Chordata', 'Craniata', 'Vertebrata', 'Euteleostomi', 'Mammalia', 'Eutheria', 'Euarchontoglires', 'Primates', 'Haplorrhini', 'Catarrhini', 'Hominidae', 'Homo']\n",
      ">references: [Reference(title='The DNA sequence and biological annotation of human chromosome 1', ...), Reference(title='Characterization of the human epidermal profilaggrin gene. Genomic organization and identification of an S-100-like calcium binding domain at the amino terminus', ...), Reference(title='Characterization of a cDNA clone encoding human filaggrin and localization of the gene to chromosome region 1q21', ...), Reference(title='Identification of the amino terminus of human filaggrin using differential LC/MS techniques: implications for profilaggrin processing', ...), Reference(title='Human Basal Tear Peptidome Characterization by CID, HCD, and ETD Followed by in Silico and in Vitro Analyses for Antimicrobial Peptide Identification', ...), Reference(title='Preferential deimination of keratin K1 and filaggrin during the terminal differentiation of human epidermis', ...), Reference(title='Loss-of-function variations within the filaggrin gene predispose for atopic dermatitis with allergic sensitizations', ...), Reference(title='Filaggrin loss-of-function mutations predispose to phenotypes involved in the atopic march', ...), Reference(title='Loss-of-function mutations in the gene encoding filaggrin cause ichthyosis vulgaris', ...), Reference(title='Common loss-of-function variants of the epidermal barrier protein filaggrin are a major predisposing factor for atopic dermatitis', ...), Reference(title='Unique mutations in the filaggrin gene in Japanese patients with ichthyosis vulgaris and atopic dermatitis', ...), Reference(title='Molecular identification and expression analysis of filaggrin-2, a member of the S100 fused-type protein family', ...), Reference(title='Deimination of human filaggrin-2 promotes its proteolysis by calpain 1', ...), Reference(title='An enzyme assisted RP-RPLC approach for in-depth analysis of human liver phosphoproteome', ...)]\n",
      ">comment: On or before Sep 26, 2008 this sequence version replaced\n",
      "gi:74720133, gi:74762958, gi:74756325, gi:1346008.\n",
      "[FUNCTION] Aggregates keratin intermediate filaments and promotes\n",
      "disulfide-bond formation among the intermediate filaments during\n",
      "terminal differentiation of mammalian epidermis.\n",
      "[INTERACTION] P20930; Q9Y337: KLK5; NbExp=4; IntAct=EBI-1058782,\n",
      "EBI-9057524.\n",
      "[SUBCELLULAR LOCATION] Cytoplasmic granule\n",
      "{ECO:0000269|PubMed:16710414}. Note=In the stratum granulosum of\n",
      "the epidermis, localized within keratohyalin granules\n",
      "(PubMed:1429717). In granular keratinocytes and in lower\n",
      "corneocytes, colocalizes with calpain-1/CAPN1 (PubMed:21531719).\n",
      "{ECO:0000269|PubMed:1429717, ECO:0000269|PubMed:21531719}.\n",
      "[TISSUE SPECIFICITY] Expressed in skin, thymus, stomach, tonsils,\n",
      "testis, placenta, kidney, pancreas, mammary gland, bladder,\n",
      "thyroid, salivary gland and trachea, but not detected in heart,\n",
      "brain, liver, lung, bone marrow, small intestine, spleen, prostate,\n",
      "colon, or adrenal gland (PubMed:19384417). In the skin, mainly\n",
      "expressed in stratum granulosum of the epidermis (PubMed:1429717)\n",
      "(PubMed:19384417). {ECO:0000269|PubMed:1429717,\n",
      "ECO:0000269|PubMed:19384417}.\n",
      "[INDUCTION] In cultured foreskin fibroblasts, up-regulated in\n",
      "response to Ca(2+) stimulation. {ECO:0000269|PubMed:19384417}.\n",
      "[PTM] Filaggrin is initially synthesized as a large, insoluble,\n",
      "highly phosphorylated precursor containing many tandem copies of\n",
      "324 AA, which are not separated by large linker sequences. During\n",
      "terminal differentiation it is dephosphorylated and proteolytically\n",
      "cleaved. The N-terminal of the mature protein is heterogeneous, and\n",
      "is blocked by the formation of pyroglutamate.\n",
      "[PTM] Undergoes deimination of some arginine residues\n",
      "(citrullination). {ECO:0000269|PubMed:8780679}.\n",
      "[DISEASE] Ichthyosis vulgaris (VI) [MIM:146700]: The most common\n",
      "form of ichthyosis inherited as an autosomal dominant trait. It is\n",
      "characterized by palmar hyperlinearity, keratosis pilaris and a\n",
      "fine scale that is most prominent over the lower abdomen, arms, and\n",
      "legs. Ichthyosis vulgaris is characterized histologically by absent\n",
      "or reduced keratohyalin granules in the epidermis and mild\n",
      "hyperkeratosis. The disease can be associated with frequent asthma,\n",
      "eczema or hay fever. {ECO:0000269|PubMed:16444271}. Note=The\n",
      "disease is caused by variants affecting the gene represented in\n",
      "this entry.\n",
      "[DISEASE] Dermatitis atopic 2 (ATOD2) [MIM:605803]: Atopic\n",
      "dermatitis is a complex, inflammatory disease with multiple alleles\n",
      "at several loci thought to be involved in the pathogenesis. It\n",
      "commonly begins in infancy or early childhood and is characterized\n",
      "by a chronic relapsing form of skin inflammation, a disturbance of\n",
      "epidermal barrier function that culminates in dry skin, and\n",
      "IgE-mediated sensitization to food and environmental allergens. It\n",
      "is manifested by lichenification, excoriation, and crusting, mainly\n",
      "on the flexural surfaces of the elbow and knee.\n",
      "{ECO:0000269|PubMed:16550169, ECO:0000269|PubMed:16815158,\n",
      "ECO:0000269|PubMed:17030239, ECO:0000269|PubMed:17291859}.\n",
      "Note=Disease susceptibility is associated with variants affecting\n",
      "the gene represented in this entry.\n",
      "[SIMILARITY] Belongs to the S100-fused protein family.\n",
      "{ECO:0000305}.\n",
      ">molecule_type: protein\n"
     ]
    }
   ],
   "source": [
    "seq_annotations(access_ncbi_seq(\"84028206\", \"protein\", \"gb\", \"flg_protein_uniprot\")) # annotations flg_protein_uniprot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Features e qualifiers da sequência da proteína do registo UniProt no NCBI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "225 features\n",
      "\n",
      "Type and Location:\n",
      ">source, [0:4061]\n",
      ">gene, [0:4061]\n",
      ">Protein, [0:4061]\n",
      ">Region, [0:4061]\n",
      ">Region, [1:88]\n",
      ">Site, order{[1:19], [24:27], [35:37], [38:41], [68:72], [73:75], [76:80], [81:88]}\n",
      ">Region, [3:19]\n",
      ">Region, [5:43]\n",
      ">Site, order{[18:19], [23:24], [26:27], [31:33], [61:62], [63:64], [65:66], [67:68], [69:70], [72:73]}\n",
      ">Region, [29:39]\n",
      ">Region, [42:46]\n",
      ">Region, [48:84]\n",
      ">Region, [49:53]\n",
      ">Region, [53:61]\n",
      ">Region, [70:85]\n",
      ">Region, [91:212]\n",
      ">Region, [108:212]\n",
      ">Region, [185:216]\n",
      ">Region, [254:3971]\n",
      ">Region, [257:306]\n",
      ">Region, [262:285]\n",
      ">Region, [285:355]\n",
      ">Region, [331:332]\n",
      ">Region, [353:>590]\n",
      ">Region, [355:452]\n",
      ">Region, [372:428]\n",
      ">Region, [373:428]\n",
      ">Region, [443:444]\n",
      ">Region, [453:454]\n",
      ">Region, [459:494]\n",
      ">Region, [477:478]\n",
      ">Region, [526:574]\n",
      ">Region, [564:>950]\n",
      ">Region, [574:595]\n",
      ">Region, [578:630]\n",
      ">Region, [595:627]\n",
      ">Region, [647:679]\n",
      ">Region, [679:700]\n",
      ">Region, [696:753]\n",
      ">Region, [697:753]\n",
      ">Region, [700:715]\n",
      ">Region, [715:750]\n",
      ">Region, [724:725]\n",
      ">Region, [741:742]\n",
      ">Region, [758:773]\n",
      ">Region, [773:788]\n",
      ">Region, [788:819]\n",
      ">Region, [842:869]\n",
      ">Region, [869:895]\n",
      ">Region, [895:921]\n",
      ">Region, [898:955]\n",
      ">Region, [903:955]\n",
      ">Region, [921:950]\n",
      ">Region, [950:965]\n",
      ">Region, [972:998]\n",
      ">Region, [984:>1368]\n",
      ">Region, [998:1025]\n",
      ">Region, [1021:1077]\n",
      ">Region, [1022:1077]\n",
      ">Region, [1025:1040]\n",
      ">Region, [1058:1074]\n",
      ">Region, [1082:1127]\n",
      ">Region, [1127:1157]\n",
      ">Region, [1165:1193]\n",
      ">Region, [1183:1184]\n",
      ">Region, [1193:1216]\n",
      ">Region, [1216:1244]\n",
      ">Region, [1224:1279]\n",
      ">Region, [1227:1279]\n",
      ">Region, [1244:1270]\n",
      ">Region, [1270:1289]\n",
      ">Region, [1295:1328]\n",
      ">Region, [1328:1349]\n",
      ">Region, [1345:1401]\n",
      ">Region, [1346:1401]\n",
      ">Region, [1349:1364]\n",
      ">Region, [1375:1376]\n",
      ">Region, [1382:1470]\n",
      ">Region, [1436:1437]\n",
      ">Region, [1481:1482]\n",
      ">Region, [1517:1547]\n",
      ">Region, [1546:1603]\n",
      ">Region, [1547:1568]\n",
      ">Region, [1551:1603]\n",
      ">Region, [1568:1591]\n",
      ">Region, [1591:1652]\n",
      ">Region, [1606:>1826]\n",
      ">Region, [1652:1676]\n",
      ">Region, [1670:1725]\n",
      ">Region, [1670:1725]\n",
      ">Region, [1683:1684]\n",
      ">Region, [1687:1722]\n",
      ">Region, [1698:1699]\n",
      ">Region, [1730:1790]\n",
      ">Region, [1749:1750]\n",
      ">Region, [1790:1805]\n",
      ">Region, [1804:1805]\n",
      ">Region, [1815:1816]\n",
      ">Region, [1823:1843]\n",
      ">Region, [1850:1871]\n",
      ">Region, [1871:1893]\n",
      ">Region, [1875:1927]\n",
      ">Region, [1890:1891]\n",
      ">Region, [1893:1924]\n",
      ">Region, [1952:1976]\n",
      ">Region, [1960:1961]\n",
      ">Region, [1976:1997]\n",
      ">Region, [1993:2050]\n",
      ">Region, [1994:2050]\n",
      ">Region, [1997:2012]\n",
      ">Region, [2021:2022]\n",
      ">Region, [2032:2047]\n",
      ">Region, [2053:2138]\n",
      ">Region, [2107:2108]\n",
      ">Region, [2118:2119]\n",
      ">Region, [2166:2189]\n",
      ">Region, [2189:2217]\n",
      ">Region, [2193:2194]\n",
      ">Region, [2196:2252]\n",
      ">Region, [2200:2252]\n",
      ">Region, [2217:2247]\n",
      ">Region, [2232:>2533]\n",
      ">Region, [2247:2301]\n",
      ">Region, [2301:2322]\n",
      ">Region, [2318:2374]\n",
      ">Region, [2319:2374]\n",
      ">Region, [2322:2337]\n",
      ">Region, [2355:2371]\n",
      ">Region, [2379:2440]\n",
      ">Region, [2443:2444]\n",
      ">Region, [2454:2476]\n",
      ">Region, [2465:2466]\n",
      ">Region, [2490:2520]\n",
      ">Region, [2506:2507]\n",
      ">Region, [2520:2541]\n",
      ">Region, [2524:2576]\n",
      ">Region, [2539:2540]\n",
      ">Region, [2541:2573]\n",
      ">Region, [2544:2545]\n",
      ">Region, [2593:2619]\n",
      ">Region, [2619:2646]\n",
      ">Region, [2642:2698]\n",
      ">Region, [2643:2698]\n",
      ">Region, [2646:2661]\n",
      ">Region, [<2648:>2961]\n",
      ">Region, [2651:2652]\n",
      ">Region, [2680:2695]\n",
      ">Region, [2703:2764]\n",
      ">Region, [2780:2781]\n",
      ">Region, [2795:2811]\n",
      ">Region, [2803:2804]\n",
      ">Region, [2811:2844]\n",
      ">Region, [2844:2866]\n",
      ">Region, [2848:2900]\n",
      ">Region, [2866:2895]\n",
      ">Region, [2895:2910]\n",
      ">Region, [2917:2943]\n",
      ">Region, [2943:2970]\n",
      ">Region, [2966:3022]\n",
      ">Region, [2967:3022]\n",
      ">Region, [2970:2985]\n",
      ">Region, [<2972:>3252]\n",
      ">Region, [3004:3019]\n",
      ">Region, [3027:3088]\n",
      ">Region, [3104:3105]\n",
      ">Region, [3119:3135]\n",
      ">Region, [3135:3168]\n",
      ">Region, [3168:3190]\n",
      ">Region, [3172:3224]\n",
      ">Region, [3178:3179]\n",
      ">Region, [3190:3219]\n",
      ">Region, [3219:3234]\n",
      ">Region, [3241:3276]\n",
      ">Region, [3276:3294]\n",
      ">Region, [3290:3346]\n",
      ">Region, [3291:3346]\n",
      ">Region, [3294:3309]\n",
      ">Region, [3327:3343]\n",
      ">Region, [3351:3367]\n",
      ">Region, [3370:3371]\n",
      ">Region, [3381:3396]\n",
      ">Region, [3395:3396]\n",
      ">Region, [3396:3426]\n",
      ">Region, [3414:3415]\n",
      ">Region, [3426:3427]\n",
      ">Region, [3435:3436]\n",
      ">Region, [3436:3437]\n",
      ">Region, [3447:3462]\n",
      ">Region, [3462:3492]\n",
      ">Region, [3489:3490]\n",
      ">Region, [3492:3513]\n",
      ">Region, [3496:3548]\n",
      ">Region, [3502:3503]\n",
      ">Region, [3511:3512]\n",
      ">Region, [3513:3545]\n",
      ">Region, [3563:3564]\n",
      ">Region, [3565:3597]\n",
      ">Region, [3583:3584]\n",
      ">Region, [3592:3593]\n",
      ">Region, [3597:3618]\n",
      ">Region, [3605:>3832]\n",
      ">Region, [3614:3670]\n",
      ">Region, [3615:3670]\n",
      ">Region, [3618:3633]\n",
      ">Region, [3629:3630]\n",
      ">Region, [3651:3667]\n",
      ">Region, [3675:3726]\n",
      ">Region, [3694:3695]\n",
      ">Region, [3695:3696]\n",
      ">Region, [3719:3720]\n",
      ">Region, [3738:3739]\n",
      ">Region, [3750:3751]\n",
      ">Region, [3759:3760]\n",
      ">Region, [3760:3761]\n",
      ">Region, [3786:3809]\n",
      ">Region, [3809:3837]\n",
      ">Region, [3813:3814]\n",
      ">Region, [3820:3872]\n",
      ">Region, [3826:3827]\n",
      ">Region, [3837:3869]\n",
      ">Region, [3870:3907]\n",
      ">Region, [3907:3967]\n",
      ">Region, [3907:3908]\n",
      ">Region, [3934:3935]\n",
      ">Region, [3969:3970]\n",
      "\n",
      "Qualifiers:\n",
      ">OrderedDict([('organism', ['Homo sapiens']), ('db_xref', ['taxon:9606'])])\n",
      ">OrderedDict([('gene', ['FLG'])])\n",
      ">OrderedDict([('product', ['Filaggrin']), ('UniProtKB_evidence', ['Evidence at protein level'])])\n",
      ">OrderedDict([('region_name', ['Mature chain']), ('note', ['Filaggrin. /id=PRO_0000144036.'])])\n",
      ">OrderedDict([('region_name', ['S-100']), ('note', ['S-100 domain, which represents the largest family within the superfamily of proteins carrying the Ca-binding EF-hand motif. Note that this S-100 hierarchy contains only S-100 EF-hand domains, other EF-hands have been modeled separately. S100 proteins are...; cd00213']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('site_type', ['other']), ('note', ['dimerization interface [polypeptide binding]']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('region_name', ['Helical region']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Domain']), ('note', ['EF-hand 1. /evidence=ECO:0000255|PROSITE-ProRule:PRU00448.'])])\n",
      ">OrderedDict([('site_type', ['other']), ('note', ['Ca2+ binding site [ion binding]']), ('db_xref', ['CDD:238131'])])\n",
      ">OrderedDict([('region_name', ['Helical region']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Hydrogen bonded turn']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Domain']), ('note', ['EF-hand 2. /evidence=ECO:0000255|PROSITE-ProRule:PRU00448.'])])\n",
      ">OrderedDict([('region_name', ['Hydrogen bonded turn']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Helical region']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Helical region']), ('note', ['/evidence=ECO:0007829|PDB:4PCW.'])])\n",
      ">OrderedDict([('region_name', ['Region of interest in the sequence']), ('note', ['Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Coiled-coil region']), ('note', ['/evidence=ECO:0000255.'])])\n",
      ">OrderedDict([('region_name', ['Region of interest in the sequence']), ('note', ['Disordered. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 1.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> V (in dbSNP:rs41267154). /id=VAR_061049.'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 2.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> R (in dbSNP:rs11588170). /id=VAR_061050.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['T -> A (in dbSNP:rs2011331). /id=VAR_059155.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['P -> S (in dbSNP:rs11584340). /id=VAR_059156.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 3.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 4.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['T -> I (in dbSNP:rs3120655). /id=VAR_059157.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> Y (in dbSNP:rs3120654). /id=VAR_061051.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 5.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 6.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> L (in dbSNP:rs3120649). /id=VAR_045968.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 7.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 8.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> G (in dbSNP:rs11581433). /id=VAR_045969.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> C (in dbSNP:rs12750571). /id=VAR_045970.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> Y (in dbSNP:rs11204978). /id=VAR_059158.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 9.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 10.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> H (in dbSNP:rs12407807). /id=VAR_061052.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> C (in dbSNP:rs12405278). /id=VAR_059159.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> F (in dbSNP:rs3120647). /id=VAR_059160.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['A -> V (in dbSNP:rs12405241). /id=VAR_045971.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Q (in dbSNP:rs12073613). /id=VAR_059161.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 11.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> Q (in dbSNP:rs12407748). /id=VAR_059162.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Q (in dbSNP:rs3126079). /id=VAR_045972.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 12.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['I -> T (in dbSNP:rs142592778). /id=VAR_045973.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['A -> V (in dbSNP:rs7522925). /id=VAR_059163.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['Y -> H (in dbSNP:rs7512553). /id=VAR_059164.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['Y -> H (in dbSNP:rs2184953). /id=VAR_059165.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 13.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['PHA03307']), ('note', ['transcriptional regulator ICP4; Provisional']), ('db_xref', ['CDD:223039'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 14.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Conflict']), ('note', ['K -> Q (in Ref. 3; AAA52454). /evidence=ECO:0000305.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Conflict']), ('note', ['P -> R (in Ref. 3; AAA52454). /evidence=ECO:0000305.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Q (in dbSNP:rs3126074). /id=VAR_045974.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 15.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> Q (in dbSNP:rs148050570). /id=VAR_048472.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> R (in dbSNP:rs3126072). /id=VAR_059166.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 16.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['ser_rich_anae_1']), ('note', ['serine-rich protein; NF033849']), ('db_xref', ['CDD:411418'])])\n",
      ">OrderedDict([('region_name', ['Conflict']), ('note', ['E -> D (in Ref. 3; AAA52454). /evidence=ECO:0000305.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['D -> Y (in dbSNP:rs759244716). /id=VAR_048473.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Conflict']), ('note', ['H -> Q (in Ref. 3; AAA52454). /evidence=ECO:0000305.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 17.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 18.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['ser_rich_anae_1']), ('note', ['serine-rich protein; NF033849']), ('db_xref', ['CDD:411418'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['Y -> D (in dbSNP:rs2065958). /id=VAR_059167.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 19.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['V -> G (in dbSNP:rs2065957). /id=VAR_059168.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 20.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> F (in dbSNP:rs3120647). /id=VAR_048474.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> P (in dbSNP:rs528344105). /id=VAR_048475.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Y (in dbSNP:rs7512553). /id=VAR_048476.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> Y (in dbSNP:rs11204978). /id=VAR_048477.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> A (in dbSNP:rs2065955). /id=VAR_033931.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Q (in dbSNP:rs12073613). /id=VAR_048478.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> C (in dbSNP:rs113933537). /id=VAR_048479.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 21.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['W -> G (in dbSNP:rs12728908). /id=VAR_059169.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['Q -> R (in dbSNP:rs571269174). /id=VAR_048480.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> H (in dbSNP:rs7518080). /id=VAR_059170.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['D -> N (in dbSNP:rs3814300). /id=VAR_048481.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['E -> D (in dbSNP:rs12083389). /id=VAR_059171.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['PRK12678']), ('note', ['transcription termination factor Rho; Provisional']), ('db_xref', ['CDD:237171'])])\n",
      ">OrderedDict([('region_name', ['Filaggrin']), ('note', ['pfam03516']), ('db_xref', ['CDD:427343'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 22.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Y (in dbSNP:rs9436065). /id=VAR_059172.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> F (in dbSNP:rs3120647). /id=VAR_048482.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['T -> A (in dbSNP:rs537909579). /id=VAR_048483.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> P (in dbSNP:rs11584340). /id=VAR_048484.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Y (in dbSNP:rs7512553). /id=VAR_048485.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> Y (in dbSNP:rs776603551). /id=VAR_048486.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> A (in dbSNP:rs768192328). /id=VAR_048487.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['H -> Q (in dbSNP:rs755367746). /id=VAR_048488.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['R -> C (in dbSNP:rs146212122). /id=VAR_048489.'])])\n",
      ">OrderedDict([('region_name', ['Repetitive region']), ('note', ['Filaggrin 23.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['G -> W (in dbSNP:rs140464988). /id=VAR_048490.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Basic and acidic residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Compositionally biased region']), ('note', ['Polar residues. /evidence=ECO:0000256|SAM:MobiDB-lite.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['D -> N (in dbSNP:rs3814300). /id=VAR_048491.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> P (in dbSNP:rs3126065). /id=VAR_048492.'])])\n",
      ">OrderedDict([('region_name', ['Variant']), ('note', ['S -> L (in dbSNP:rs3814299). /id=VAR_048493.'])])\n"
     ]
    }
   ],
   "source": [
    "seq_features_qualifiers(access_ncbi_seq(\"84028206\", \"protein\", \"gb\", \"flg_protein_uniprot\")) # features, qualifiers flg_protein_uniprot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# acrescentar tudo sobre o gene C11ORF30 seguindo a ordem anterior\n",
    "# acrescentar tudo sobre o gene TSLP seguindo a ordem anterior"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
